res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js

```diff
@@ -0,0 +1,927 @@
+    engine.softTakeover("[EqualizerRack1_[Channel1]_Effect1]", "parameter1", true);
+
+    engine.softTakeover("[EqualizerRack1_[Channel1]_Effect1]", "parameter3", true);
+    engine.softTakeover("[EqualizerRack1_[Channel2]_Effect1]", "parameter2", true);
+    engine.softTakeover("[EqualizerRack1_[Channel3]_Effect1]", "parameter1", true);
+
+    engine.softTakeover("[QuickEffectRack1_[Channel1]]", "super1", true);
+    engine.softTakeover("[QuickEffectRack1_[Channel2]]", "super1", true);
+
+    engine.softTakeover("[Master]", "crossfader", true);
+    engine.softTakeover("[Master]", "gain", true);
+    engine.softTakeover("[Master]", "headMix", true);
+    engine.softTakeover("[Master]", "headGain", true);
+
+    for (var i = 1; i <= 16; ++i) {
+        engine.softTakeover("[Sampler" + i + "]", "pregain", true);
+    }
+
+    // Dirty hack to set initial values in the packet parser
+    var data = [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+    TraktorS2MK3.incomingData(data);
+};
+
+TraktorS2MK3.registerInputJog = function (message, group, name, offset, bitmask, callback) {
+    // Jog wheels have 4 byte input
+    message.addControl(group, name, offset, "I", bitmask);
+    message.setCallback(group, name, callback);
+};
+
+TraktorS2MK3.registerInputScaler = function (message, group, name, offset, bitmask, callback) {
+    message.addControl(group, name, offset, "H", bitmask);
+    message.setCallback(group, name, callback);
+};
+
+TraktorS2MK3.registerInputButton = function (message, group, name, offset, bitmask, callback) {
+    message.addControl(group, name, offset, "B", bitmask);
+    message.setCallback(group, name, callback);
+};
+
+TraktorS2MK3.playHandler = function (field) {
+    if (TraktorS2MK3.shiftPressed[field.group]) {
+        engine.setValue(field.group, "start_stop", field.value);
+    } else if (field.value === 1) {
+        script.toggleControl(field.group, "play");
+    }
+};
+
+TraktorS2MK3.cueHandler = function (field) {
+    if (TraktorS2MK3.shiftPressed[field.group]) {
+        engine.setValue(field.group, "cue_gotoandstop", field.value);
+    } else {
+        engine.setValue(field.group, "cue_default", field.value);
+    }
+};
+
+TraktorS2MK3.shiftHandler = function (field) {
+    TraktorS2MK3.shiftPressed[field.group] = field.value;
+    engine.setValue("[Controls]", "touch_shift", field.value);
+    TraktorS2MK3.outputHandler(field.value, field.group, "shift");
+};
+
+TraktorS2MK3.keylockHandler = function (field) {
+    if (field.value === 0) {
+        return;
+    }
+
+    script.toggleControl(field.group, "keylock");
+};
+
+TraktorS2MK3.syncHandler = function (field) {
+    if (TraktorS2MK3.shiftPressed[field.group]) {
+        engine.setValue(field.group, "beatsync_phase", field.value);
+        // Light LED while pressed
+        TraktorS2MK3.outputHandler(field.value, field.group, "sync_enabled");
+    } else {
+        script.toggleControl(field.group, "beatsync");
```

Comment:
Move this below into the `if (field.value)` block and use `script.triggerControl`

Reply:
Err wait, that still isn't quite right. Take a look how this is [implemented](https://github.com/mixxxdj/mixxx/blob/master/res/controllers/midi-components-0.0.js#L255) in the Components library for MIDI devices

Reply:
I think it was already working correct before, but i updated the implementation like in the Components library. https://github.com/mi01/mixxx/commit/9aaf3afcfe4435b1ebd09663d4c6b44f9bdcb090